lib/logstash/filters/dns.rb

```diff
@@ -122,6 +122,12 @@ def resolve(event)
     @resolve.each do |field|
       is_array = false
       raw = event.get(field)
+
+      if raw.nil?
+        @logger.warn("DNS filter could not resolve missing field", :field => field)
```

Comment:
My mistake, I didn't realize you were referring to the fact that the contents of the field might contain a nil. I didn't realize this plugin did that as well.

I don't think we should stop because of a failure, I think we should add a `tag_on_failure` feature like grok. 

I do think that is out of scope for this PR because failures can be caused by both NPEs and DNS errors. The scope of this PR IMHO is addressing NPEs.